processing/src/main/java/io/druid/segment/incremental/IncrementalIndexAdapter.java

```diff
@@ -186,18 +186,18 @@ public int indexOf(String value)
          */
         return (Iterators.transform(
             index.getFacts().entrySet().iterator(),
-            new Function<Map.Entry<IncrementalIndex.TimeAndDims, Integer>, Rowboat>()
+            new Function<Map.Entry<IncrementalIndex.TimeAndDims, ?>, Rowboat>()
             {
               int count = 0;
 
               @Override
               public Rowboat apply(
-                  @Nullable Map.Entry<IncrementalIndex.TimeAndDims, Integer> input
+                  @Nullable Map.Entry<IncrementalIndex.TimeAndDims, ?> input
               )
               {
                 final IncrementalIndex.TimeAndDims timeAndDims = input.getKey();
                 final String[][] dimValues = timeAndDims.getDims();
-                final int rowOffset = input.getValue();
+                final Object aggrs = input.getValue();
```

Comment:
is there no way to have some kind of type information here, using Object makes the code hard to read. We might also want to use more explicit variable names if we have to remove type information.